tests/Resources/modules/ps_banner/ps_banner.php

```diff
@@ -0,0 +1,265 @@
+        $helper->default_form_language = $lang->id;
+        $helper->module = $this;
+        $helper->allow_employee_form_lang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') ? Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') : 0;
+        $helper->identifier = $this->identifier;
+        $helper->submit_action = 'submitStoreConf';
+        $helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false).'&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name;
+        $helper->token = Tools::getAdminTokenLite('AdminModules');
+        $helper->tpl_vars = array(
+            'uri' => $this->getPathUri(),
+            'fields_value' => $this->getConfigFieldsValues(),
+            'languages' => $this->context->controller->getLanguages(),
+            'id_language' => $this->context->language->id
+        );
+
+        return $helper->generateForm(array($fields_form));
+    }
+
+    public function getConfigFieldsValues()
+    {
+        $languages = Language::getLanguages(false);
+        $fields = array();
+
+        foreach ($languages as $lang) {
+            $fields['BANNER_IMG'][$lang['id_lang']] = Tools::getValue('BANNER_IMG_'.$lang['id_lang'], Configuration::get('BANNER_IMG', $lang['id_lang']));
+            $fields['BANNER_LINK'][$lang['id_lang']] = Tools::getValue('BANNER_LINK_'.$lang['id_lang'], Configuration::get('BANNER_LINK', $lang['id_lang']));
+            $fields['BANNER_DESC'][$lang['id_lang']] = Tools::getValue('BANNER_DESC_'.$lang['id_lang'], Configuration::get('BANNER_DESC', $lang['id_lang']));
+        }
+
+        return $fields;
+    }
+
+    public function renderWidget($hookName, array $params)
+    {
+        if (!$this->isCached($this->templateFile, $this->getCacheId('ps_banner'))) {
+            $this->smarty->assign($this->getWidgetVariables($hookName, $params));
+        }
+
+        return $this->fetch($this->templateFile, $this->getCacheId('ps_banner'));
+    }
+
+    public function getWidgetVariables($hookName, array $params)
+    {
+        $imgname = Configuration::get('BANNER_IMG', $this->context->language->id);
+
+        if ($imgname && file_exists(_PS_MODULE_DIR_.$this->name.DIRECTORY_SEPARATOR.'img'.DIRECTORY_SEPARATOR.$imgname)) {
+            $this->smarty->assign('banner_img', $this->context->link->protocol_content . Tools::getMediaServer($imgname) . $this->_path . 'img/' . $imgname);
+        }
+
+        $banner_link = Configuration::get('BANNER_LINK', $this->context->language->id);
+        if (!$banner_link) {
+            $banner_link = $this->context->link->getPageLink('index');
+        }
+
+        return array(
+            'banner_link' => $this->updateUrl($banner_link),
+            'banner_desc' => Configuration::get('BANNER_DESC', $this->context->language->id)
+        );
+    }
+
+    private function updateUrl($link)
+    {
+        if (substr($link, 0, 7) !== "http://" && substr($link, 0, 8) !== "https://") {
```

Comment:
easier to use: `parse_url($url, PHP_URL_SCHEME)`
Plus, better to retrieve the `Tools::getShopProtocol`?

Reply:
lol, just realized it's in the test directory xD

Reply:
Yep, I thought you did it on purpose ^^

Reply:
Little bit tired today -_-'